[General] Generic Azerite Proc Card + Synergistic Growth Azerite Trait

TJMoats opened the pull request:
I've done three things here, first off I've created a GenericStatProc class for any Azerite traits that just have a "give you x stat for y seconds". I based my design off of the existing BloodRite Azerite Trait.

I then took the generic class and created a tracker for Synergistic Growth out of it. You can see that here:

https://i.imgur.com/GQPe8wF.png
/report/4qXRxW1fYLPaJgMA/7-Mythic+Zul+-+Wipe+1+(1:25)/8-Monkacarl

I then ported over blood rite to also use the GenericStatProc. Nothing should have changed about the card, but here is what it looks like: 

https://i.imgur.com/Gd52rFZ.png
/report/k4bAJZKWVaGt12j9/3-Mythic+Taloc+-+Kill+(4:51)/14-Cistara

I wasn't 100% sure where to put the GenericStatProc, so I just put it in with the azerite traits.

Pewtro commented on src/parser/core/modules/spells/bfa/azeritetraits/SynergisticGrowth.js:
```diff
@@ -0,0 +1,21 @@
+export const STAT_TRACKER = {
+  mastery: combatant => azeriteStats(combatant.traitsBySpellId[SPELLS.BLOOD_RITE.id]).stat,
+};
+
```
Need to update this.

Pewtro commented on src/parser/core/modules/spells/bfa/azeritetraits/SynergisticGrowth.js:
```diff
@@ -0,0 +1,21 @@
+  constructor(...args) {
+    super(SPELLS.SYNERGISTIC_GROWTH.id, SPELLS.SYNERGISTIC_GROWTH_BUFF.id, STAT.MASTERY, ...args);
+  }
+}
```
As far as I remember these buffs are currently in a resto druid module from this PR https://github.com/WoWAnalyzer/WoWAnalyzer/pull/2235/files#diff-d4d8abcfdfba4098a816011f1545e4dd

Can we have them moved out

TJMoats replied:
Huh, I didn't realize they had added it. Should we check for other general traits that are implemented for specific classes?

Pewtro replied:
Wouldn't hurt to do a global search for the spell IDs when implementing, just to double check

TJMoats commented:
I've removed the resto druid specific trait and fixed the typo. 

Old Druid Card: https://puu.sh/BEwPY/d9a5e5152c.png
New Druid Card: https://puu.sh/BEwMr/bd81a48dee.png
